### PR TITLE
Updated resteasy version

### PR DESCRIPTION
**Pull request by Graeme-Miller**
This is successor to [this PR](https://github.com/apache/brooklyn-server/pull/317).
After the rest client was moved to brooklyn-client in [this PR](https://github.com/apache/brooklyn-client/pull/25) two new PRs had to be created, one for brooklyn-server and one for brooklyn-client.

Please merge [this PR](https://github.com/apache/brooklyn-server/pull/339) at the same time.

This PR updates resteasy to 3.0.14.Final.

### CODE REVIEW

**Comment by Graeme-Miller** [comment]
Test failures are because brooklyn-client relies on brooklyn-rest-api which pulls in an older verision of jackson.

All tests pass locally when https://github.com/apache/brooklyn-server/pull/339 is included.

**Comment by neykov** [comment]
@Graeme-Miller this change won't be needed if https://github.com/apache/brooklyn-client/pull/27 is merged.

**Comment by aledsage** [comment]
@Graeme-Miller Can you explain why we're upgrading to resteasy `3.0.14.Final`?  There is a `3.0.19.Final` available - why not upgrade to that?

---
TL;DR we don't bundle `brooklyn-rest-client` with the Brooklyn distro, so I think we can ignore the version difference discussed below.

I see that resteasy depends on `com.fasterxml.jackson` version `2.6.3` (the previous resteasy resteasy 3.0.8 depended on fasterxml version `2.3.2`).

However, in https://github.com/apache/brooklyn-server/pull/339, you're upgrading `com.fasterxml.jackson` from 2.4.5 to 2.7.5. That means mvn is trying to pull in two different versions of fasterxml. That's manageable in Karaf (though not ideal), but is a pain with the vanilla Java main. We'd need to have `<excludes>` to ensure we get the right single version.

I see that resteasy has a `3.1.0.Beta2` available, which depends on fasterxml 2.7.4. It came out on 9th September 2016; the Beta1 was on 28th July 2016. That depends on fasterxml 2.7.4. It took 6 beta releases before a 3.0.0-rc1 and then 3.0.0.Final, so we shouldn't hold our breath!

**Comment by aledsage** [comment]
LGTM. https://github.com/apache/brooklyn-server/pull/339 is merged; merging this now.
